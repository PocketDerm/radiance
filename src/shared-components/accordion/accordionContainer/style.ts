import styled from '@emotion/styled';
import { BOX_SHADOWS, BREAKPOINTS, COLORS } from 'src/constants';

const Container = styled.div<{ borderRadius?: string }>`
  border-radius: ${({ borderRadius }) => borderRadius || 'inherit'};
  box-shadow: ${BOX_SHADOWS.boxShadow4};
  background-color: ${COLORS.white};
  max-width: ${BREAKPOINTS.md}px;
`;

export default { Container };
