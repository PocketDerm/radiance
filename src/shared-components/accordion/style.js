import styled from '@emotion/styled';
import { css } from '@emotion/core';

import {
  ANIMATION,
  BOX_SHADOWS,
  BREAKPOINTS,
  COLORS,
  SPACER,
} from '../../constants';

const border = `1px solid ${COLORS.border}`;

export const Container = styled.div`
  box-shadow: ${BOX_SHADOWS.boxShadow4};
  background-color: ${COLORS.white};
  max-width: ${BREAKPOINTS.md}px;
`;

export const Content = styled.div`
  padding: ${SPACER.medium};
  width: 100%;
`;

export const ExpansionWrapper = styled.div`
  max-height: ${({ contentHeight }) => contentHeight};
  overflow: hidden;
  transition: max-height ${ANIMATION.defaultTiming} ease-in-out;
`;

const getBorderStyle = isOpen => css`
  border: ${border};

  ${ExpansionWrapper} {
    ${isOpen && `border-top: ${border};`};
  }
`;

export const AccordionBox = styled.div`
  ${({ noBorder, isOpen }) => !noBorder && getBorderStyle(isOpen)};
  width: 100%;

  &:not(:last-child) {
    border-bottom: none;
  }

  ${({ disabled }) =>
    disabled &&
    `
    opacity: 0.4;
    background-color: ${COLORS.disabled};
    border-color: ${COLORS.purple30};
  `};
`;

export const ArrowWrapper = styled.div`
  display: flex;
  align-items: center;
  padding: 0 ${SPACER.medium};
`;

export const TitleWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
`;

export const Truncate = styled.div`
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
`;
