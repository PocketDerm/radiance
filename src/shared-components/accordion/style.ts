import styled from '@emotion/styled';
import { css } from '@emotion/core';
import {
  ANIMATION,
  BREAKPOINTS,
  BOX_SHADOWS,
  COLORS,
  SPACER,
} from 'src/constants';

const border = `1px solid ${COLORS.border}`;

const getTopBorderRadius = (borderRadius: string) => ({
  borderTopLeftRadius: borderRadius,
  borderTopRightRadius: borderRadius,
});

const getBottomBorderRadius = (borderRadius: string) => ({
  borderBottomLeftRadius: borderRadius,
  borderBottomRightRadius: borderRadius,
});

export const Content = styled.div`
  padding: ${SPACER.medium};
  width: 100%;
`;

export const ExpansionWrapper = styled.div<{ contentHeight: string }>`
  max-height: ${({ contentHeight }) => contentHeight};
  overflow: hidden;
  transition: max-height ${ANIMATION.defaultTiming} ease-in-out;
`;

const getBorderStyle = (isOpen: boolean) => css`
  border: ${border};

  ${ExpansionWrapper} {
    ${isOpen && `border-top: ${border};`};
  }
`;

export const AccordionBox = styled.div<{
  noBorder: boolean;
  isOpen: boolean;
  disabled: boolean;
}>`
  ${({ noBorder, isOpen }) => (!noBorder ? getBorderStyle(isOpen) : '')};

  position: relative;

  width: 100%;

  &:not(:last-child) {
    border-bottom: none;
  }

  ${({ disabled }) =>
    disabled
      ? `
    opacity: 0.4;
    background-color: ${COLORS.disabled};
    border-color: ${COLORS.purple30};
  `
      : ''};
`;

export const ArrowWrapper = styled.div<{ rightAlign: boolean }>`
  display: flex;
  align-items: center;

  ${({ rightAlign }) =>
    rightAlign
      ? `padding-left: ${SPACER.medium};`
      : `padding: 0 ${SPACER.medium};`};
`;

export const TitleWrapper = styled.div<{
  borderRadius: string;
  disabled: boolean;
  isOpen: boolean;
}>`
  display: flex;
  justify-content: space-between;
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
  &:focus {
    outline: none;
    box-shadow: ${BOX_SHADOWS.focusSecondary};
  }

  div:last-of-type & {
    &:focus {
      ${({ borderRadius, isOpen }) => {
    const {
      borderBottomLeftRadius,
      borderBottomRightRadius,
    } = getBottomBorderRadius(borderRadius);

    return (
      !isOpen &&
          `
        border-bottom-left-radius: ${borderBottomLeftRadius}; 
        border-bottom-right-radius: ${borderBottomRightRadius};
        `
    );
  }}
    }
  }
`;

export const Truncate = styled.div`
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
`;

/**
 * borderRadius must match borderRadius passed to main <Accordion />
 * component if opting out of default values.
 */
export const Container = styled.div<{ borderRadius?: string }>`
  box-shadow: ${BOX_SHADOWS.clickable};
  background-color: ${COLORS.white};
  max-width: ${BREAKPOINTS.md}px;

  ${({ borderRadius = '4px' }) => {
    const { borderTopLeftRadius, borderTopRightRadius } = getTopBorderRadius(
      borderRadius,
    );
    const {
      borderBottomLeftRadius,
      borderBottomRightRadius,
    } = getBottomBorderRadius(borderRadius);

    return `
    > div:first-of-type {
      border-top-left-radius: ${borderTopLeftRadius};
      border-top-right-radius: ${borderTopRightRadius};

      ${TitleWrapper} {
        border-top-left-radius: ${borderTopLeftRadius};
        border-top-right-radius: ${borderTopRightRadius};
      }

      ${AccordionBox} {
        border-top-left-radius: ${borderTopLeftRadius};
        border-top-right-radius: ${borderTopRightRadius};
      }
    }
  
    > div:last-of-type {
      border-bottom-left-radius: ${borderBottomLeftRadius};
      border-bottom-right-radius: ${borderBottomRightRadius};
  
      ${AccordionBox} {
        border-bottom-left-radius: ${borderBottomLeftRadius};
        border-bottom-right-radius: ${borderBottomRightRadius};
      }
    }
`;
  }}
`;
