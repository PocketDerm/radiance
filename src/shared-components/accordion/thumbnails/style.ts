import styled from '@emotion/styled';
import { css } from '@emotion/core';

import { style as TYPOGRAPHY_STYLE } from '../../typography';
import { COLORS, SPACER } from '../../../constants';

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  margin-left: ${SPACER.large};
`;

const thumbnailBase = css`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 1.5rem;
  width: 1rem;
`;

export const ImageContainer = styled.div`
  ${thumbnailBase} overflow: hidden;
  height: 1.5rem;
  width: 1rem;

  &:not(:last-child) {
    margin-right: 4px;
  }
`;

export const ThumbnailImage = styled.img`
  min-height: 1.5rem;
  min-width: 1rem;
  object-fit: cover;
`;

export const MultiplesContainer = styled.div`
  ${thumbnailBase}
  background-color: ${COLORS.purple15};
`;

export const MultiplesText = styled.div`
  ${TYPOGRAPHY_STYLE.label};
  font-size: 10px;
`;
