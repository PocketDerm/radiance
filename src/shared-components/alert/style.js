import styled from '@emotion/styled';
import { css, keyframes } from '@emotion/core';

import {
  COLORS,
  MEDIA_QUERIES,
  SPACER,
  ANIMATION,
  TYPOGRAPHY_CONSTANTS,
} from '../../constants';

export const AlertsContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  position: fixed;
  top: ${SPACER.medium};
  left: 0;
  right: 0;
  z-index: 99999;

  ${MEDIA_QUERIES.mdUp} {
    left: auto;
    right: ${SPACER.medium};
  }
`;

const fadeInDesktop = keyframes`
  from { opacity: 0; transform: translate3d(24px, 0, 0); }
  to { opacity: 1; transform: translate3d(0, 0, 0); }
`;

const fadeInMobile = keyframes`
  from { opacity: 0; transform: translate3d(0, -24px, 0);  }
  to { opacity: 1; transform: translate3d(0, 0px, 0); }
`;

const defaultAlertStyles = css`
  background-color: ${COLORS.primary};
  box-shadow: 0px 8px 24px rgba(51, 46, 84, 0.05);
`;

const successAlertStyles = css`
  background-color: ${COLORS.success};
  box-shadow: 0px 8px 24px rgba(43, 110, 51, 0.05);
`;

const errorAlertStyles = css`
  background-color: ${COLORS.error};
  box-shadow: 0px 8px 24px rgba(189, 32, 15, 0.05);
`;

export const AlertContainer = styled.div`
  cursor: pointer;
  position: relative;
  margin: 0 auto ${SPACER.small};
  padding: 0;
  width: 327px;
  border-radius: ${SPACER.small};
  opacity: ${props => (props.exiting ? '0' : '1')};
  animation: ${fadeInMobile} ${ANIMATION.defaultTiming} 1;
  transition: ${ANIMATION.defaultTiming};
  transform: ${({ exiting }) =>
    exiting ? 'translate3d(0, -24px, 0)' : 'translate3d(0, 0, 0)'};

  ${({ alertType }) => {
    switch (alertType) {
      case 'danger':
        return errorAlertStyles;
      case 'error':
        return errorAlertStyles;
      case 'success':
        return successAlertStyles;
      default:
        return defaultAlertStyles;
    }
  }};

  ${MEDIA_QUERIES.mdUp} {
    animation: ${fadeInDesktop} ${ANIMATION.defaultTiming} 1;
    margin: 0 auto ${SPACER.medium};
    transform: ${({ exiting }) =>
      exiting ? 'translate3d(24px, 0, 0)' : 'translate3d(0, 0, 0)'};
  }
`;

export const MainContainer = styled.div`
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-start;
  align-items: flex-start;
  color: ${COLORS.white};
  font-size: ${TYPOGRAPHY_CONSTANTS.fontSize.caption};
  padding: ${SPACER.medium};
`;

export const ContentContainer = styled.div`
  margin: -3px 0 0 ${SPACER.medium};
  max-height: 48px;
`;

export const CtaContent = styled.div`
  padding: ${SPACER.medium};

  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
  color: ${COLORS.white};
  font-size: ${TYPOGRAPHY_CONSTANTS.fontSize.caption};

  ${({ alertType }) => {
    switch (alertType) {
      case 'danger':
        return css`
          border-top: 1px solid ${COLORS.errorBorder};
        `;
      case 'error':
        return css`
          border-top: 1px solid ${COLORS.errorBorder};
        `;
      case 'success':
        return css`
          border-top: 1px solid ${COLORS.successBorder};
        `;
      default:
        return css`
          border-top: 1px solid ${COLORS.infoBorder};
        `;
    }
  }};
`;

export const IconContainer = styled.div`
  ${({ hasAvatar }) =>
    hasAvatar &&
    css`
      min-width: 32px;
    `};

  svg {
    height: ${SPACER.medium};
    width: ${SPACER.medium};
    fill: ${COLORS.white};
  }
`;
