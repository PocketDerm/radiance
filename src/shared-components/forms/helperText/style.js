import styled from '@emotion/styled';
import { css } from '@emotion/core';

import { style as TYPOGRAPHY_STYLE } from '../../typography';

export const List = styled.ul`
  list-style-type: none;
  margin: 8px 0;
  min-height: 23px;
`;

export const CenteredList = styled(List)`
  text-align: center;
`;

export const ErrorItem = styled.li`
  ${TYPOGRAPHY_STYLE.error};
`;

export const HintItem = styled.li`
  ${TYPOGRAPHY_STYLE.caption};
`;

export const transitionStyles = css`
  .expand-enter {
    opacity: 0;
    max-height: 0;
  }

  .expand-enter.expand-enter-active {
    opacity: 1;
    max-height: 50px;
    transition-duration: 350ms;
    transition-property: max-height, opacity;
    transition-timing-function: ease-in;
  }

  .expand-leave {
    opacity: 1;
    max-height: 50px;
  }

  .expand-leave.expand-leave-active {
    opacity: 0;
    max-height: 0;
    transition-duration: 350ms;
    transition-property: max-height, opacity;
    transition-timing-function: ease-in;
  }
`;
