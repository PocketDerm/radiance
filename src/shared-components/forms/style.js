import styled from '@emotion/styled';
import { css } from '@emotion/core';

import { style as TYPOGRAPHY_STYLE } from '../typography';
import { COLORS, BOX_SHADOWS, SPACING, ANIMATION } from '../../constants';

export const HelperList = styled.ul`
  list-style-type: none;
  margin: 8px 0;
  min-height: 23px;
`;

export const ErrorItem = styled.li`
  ${TYPOGRAPHY_STYLE.error};
`;

export const HintItem = styled.li`
  ${TYPOGRAPHY_STYLE.caption};
`;

export const FieldContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: ${SPACING.small};
`;

const inputStyles = css`
  ${TYPOGRAPHY_STYLE.body};
  appearance: none;
  background: ${COLORS.white};
  border: 1px solid ${COLORS.border};
  border-radius: 0;
  color: ${COLORS.primaryTint1};
  padding: 13px ${SPACING.small};
  transition: border-color ${ANIMATION.defaultTiming};
  width: 100%;
  outline: none;

  ~ ul ${HintItem} {
    transition: opacity ${ANIMATION.defaultTiming};
    max-height: 0 !important;
    opacity: 0 !important;
  }

  &:active,
  &:focus {
    outline: none;
    border-color: ${COLORS.primary};
    box-shadow: ${BOX_SHADOWS.focusSecondary};

    ~ ul ${HintItem} {
      max-height: 50px !important;
      opacity: 1 !important;
    }
  }

  &::placeholder {
    color: ${COLORS.primaryTint3};
  }

  &[disabled] {
    background-color: ${COLORS.disabled};
    color: ${COLORS.textDisabled};
    cursor: not-allowed;
    border: 1px solid ${COLORS.border};
    box-shadow: none;
  }
`;

export const Input = styled.input`
  ${inputStyles};
`;

export const Textarea = styled.textarea`
  ${inputStyles};
  color: ${COLORS.primary};
  display: block;
  height: 100%;
  line-height: ${SPACING.base};
  margin: 0 auto;
  max-width: 35rem;
  padding: ${SPACING.small};
  resize: none;
  transition: border-color ${ANIMATION.defaultTiming};
  width: 100%;
`;

const errorStyles = css`
  position: relative;

  svg {
    display: block;
    width: 16px;
    height: 16px;
    position: absolute;
    margin-top: 20px;
    right: 16px;
    fill: ${COLORS.statusRed};
  }

  ${Textarea}, ${Input} {
    border-color: ${COLORS.statusRed};

    &:active,
    &:focus {
      border-color: ${COLORS.statusRed};
      box-shadow: none;
    }
  }
`;

export const InputContainer = styled.div`
  svg {
    display: none;
  }

  ${({ showError }) => showError && errorStyles};
`;
