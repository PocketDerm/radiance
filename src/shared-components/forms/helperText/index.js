import PropTypes from 'prop-types';
import React from 'react';
import { CSSTransitionGroup } from 'react-transition-group';
import { Global } from '@emotion/core';

import { CenteredList, List, transitionStyles } from './style';
import { ANIMATION } from '../../../constants';

const HelperText = ({ children, centered }) => (
  <React.Fragment>
    <Global styles={transitionStyles} />
    <CSSTransitionGroup
      component={centered ? CenteredList : List}
      transitionName="expand"
      transitionEnterTimeout={ANIMATION.defaultTiming}
      transitionLeaveTimeout={ANIMATION.defaultTiming}
    >
      {children}
    </CSSTransitionGroup>
  </React.Fragment>
);

HelperText.propTypes = {
  centered: PropTypes.bool,
  children: PropTypes.node,
};

export default HelperText;

export { ErrorItem, HintItem } from './style';
