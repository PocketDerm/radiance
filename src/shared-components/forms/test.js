import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { ErrorItem } from './helperText/style';

import Field from './index';

describe('<Field />', () => {
  describe('UI Snapshot', () => {
    it('renders with label prop', () => {
      const labelText = 'Test Label';
      const component = renderer.create(<Field label={labelText} />);

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders with label and labelFor', () => {
      const labelText = 'Test Label';
      const labelFor = 'Test For';
      const component = renderer.create(
        <Field label={labelText} labelFor={labelFor} />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when provided an errorMessage and field is invalid show the message', () => {
    it('renders the error message', () => {
      const errorMessage = 'Maximum 6 characteres';
      const wrapper = mount(
        <Field errorMessage={errorMessage} isValid={false} />
      );
      const li = wrapper.find(ErrorItem);

      expect(li).toHaveLength(1);
      expect(li.text().match(errorMessage)).toBeTruthy();
    });
  });
});
