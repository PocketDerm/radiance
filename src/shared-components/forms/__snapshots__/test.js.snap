// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Field /> UI Snapshot renders with label and labelFor 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 1rem;
}

.emotion-0 {
  color: #5C5876;
  font-size: 0.75rem;
  line-height: 1.67;
}

.emotion-4 svg {
  display: none;
}

.emotion-2 {
  list-style-type: none;
  margin: 8px 0;
  min-height: 23px;
}

<div
  className="emotion-6 emotion-7"
>
  <label
    className="emotion-0 emotion-1"
    htmlFor="Test For"
  >
    Test Label
  </label>
  <div
    className="emotion-4 emotion-5"
  >
    <svg />
    <ul
      className="emotion-2 emotion-3"
    />
  </div>
</div>
`;

exports[`<Field /> UI Snapshot renders with label prop 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 1rem;
}

.emotion-0 {
  color: #5C5876;
  font-size: 0.75rem;
  line-height: 1.67;
}

.emotion-4 svg {
  display: none;
}

.emotion-2 {
  list-style-type: none;
  margin: 8px 0;
  min-height: 23px;
}

<div
  className="emotion-6 emotion-7"
>
  <label
    className="emotion-0 emotion-1"
    htmlFor="Test Label"
  >
    Test Label
  </label>
  <div
    className="emotion-4 emotion-5"
  >
    <svg />
    <ul
      className="emotion-2 emotion-3"
    />
  </div>
</div>
`;
